first-aid.lic

```diff
@@ -8,24 +8,43 @@ class FirstAid
   include DRC
 
   def initialize
+    settings = get_settings
     @chart_data = get_data('anatomy-charts').first_aid_charts
-
-    train_first_aid
+    if settings.textbook
+      @booktype = 'textbook'
+      textbook_charts
+    else
+      @booktype = 'compendium'
+      compendium_charts
+    end
   end
-
-  def train_first_aid
+  def compendium_charts
     bput('get my compendium', 'You get')
     bput('look my compendium', '^The compendium lies open to the section on .* physiology')
     pause
-    charts = reget(40).grep(/^   .+$/).map(&:strip)
-
-    charts_to_read = charts.select { |chart| @chart_data[chart]['scholarship'] <= DRSkill.getrank('Scholarship') }
-
-    charts_to_read.each do |chart|
-      bput("turn my compendium to #{@chart_data[chart]['index']}", 'You turn')
-      bput('study my compendium', 'you begin to study', 'you begin studying', 'With a sudden moment of', 'In a sudden moment of clarity', '^Why ', 'you continue to study', 'you attempt', 'you continue studying')
-      waitrt?
-      break if DRSkill.getxp('First Aid') >= 30
+    charts_in_compendium = reget(40).grep(/^   .+$/).map(&:strip)
+    charts_to_read = @chart_data.select { |chart, info| info['scholarship'] <= DRSkill.getrank('Scholarship') }
+                                .select { |chart, info| charts_in_compendium.include?(info['index']) }
```

Comment:
Ahh I see what you are saying.  Just for that line:

.select { |name, info| compendium_charts.include?(info['index']) }

I think this basically got by working regardless because it's using include and index is always included in name...  So even if compendium has blue-bellies crocodile - index is blue-bellied and for an include? that's enough